internal/animation/animation.go

```diff
@@ -13,11 +14,28 @@ type anim struct {
 	reverse     bool
 	start       time.Time
 	total       int64
+
+	mu      sync.RWMutex
+	stopped bool
 }
 
 func newAnim(a *fyne.Animation) *anim {
+	// TODO should we add a nil check here? (to avoid panic)
 	animate := &anim{a: a, start: time.Now(), end: time.Now().Add(a.Duration)}
 	animate.total = animate.end.Sub(animate.start).Nanoseconds() / 1000000 // TODO change this to Milliseconds() when we drop Go 1.12
 	animate.repeatsLeft = a.RepeatCount
 	return animate
 }
+
+func (a *anim) setStopFlag() {
```

Comment:
It would be more symmetrical as `setStopped`

Reply:
Agreed. I will update the name.

Reply:
done